packages/calypso-e2e/src/lib/components/comments-component.ts

```diff
@@ -26,14 +26,22 @@ export class CommentsComponent extends BaseContainer {
 	 * @returns {Promise<void>} No return value.
 	 */
 	async postComment( comment: string ): Promise< void > {
-		// Wait for the page to fully load. Otherwise, the Post Comment button may not
+		// Wait for all network connections to complete. Otherwise, the Post Comment button may not
 		// appear even if the text area is clicked on.
 		await this.page.waitForLoadState( 'networkidle' );
+		// Wait until the comment text area is fully stable on the page.
+		// This is to guard against long-loading pages (eg. notifications test) where all network
+		// requests may have completed but the page remains in a loading state.
+		const commentArea = await this.page.waitForSelector( selectors.commentTextArea );
+		await commentArea.waitForElementState( 'stable' );
```

Comment:
I've written a longish explanation why this is necessary in the description of this PR.

Reply:
I'm wondering how this actually fixes the issue. Reading the docs, the element is considered stable if "it has maintained the same bounding box for at least two consecutive animation frames". Do you think it's possible for the textarea element - because it's not animated itself IIUC - to maintain the bounding box for 2 frames during the page content load and thus lead to a flaky behavior?

Reply:
This change hasn't introduced flakiness in my local and CI runs so far. I have found (while migrating this test case) that it was the absence of this change that in fact led to flaky behavior, even locally. 

If you'd like to test this out, try this:

- pull branch, transpile, the standard things
- open `comments-component.ts`
- comment out Line 35-36
- run `DEBUG=pw:api yarn jest --detectOpenHandles --verbose --testTimeout 999999999 specs/specs-playwright/wp-notifications__trash-spec.js`

In my experience this results in the test flaking out about half the time.

Now, I don't know exactly why this addition resolve the issue. I suspect it might have to do with the fact that even after the page load is supposedly 'finished', the comments list continues to load new content and so the comment text area is shifted down continuously until all comments are done loading. This is just my (uneducated) hypothesis however.

Reply:
I did some more digging and observed the following from when the target post was clicked on to the step immediately prior to leaving a comment:

```
 pw:api   waiting for scheduled navigations to finish +0ms
  pw:api   navigated to "p7Cmog-12-p2" +2s
  pw:api   navigated to "p7Cmog-12-p2" +6ms
  pw:api   navigated to "p7Cmog-12-p2" +16ms
  pw:api   navigations have finished +1ms
  pw:api <= page.click succeeded +4ms
  pw:api   "domcontentloaded" event fired +559ms
  pw:api   "load" event fired +1ms
  pw:api navigating to "https://playwright/index.html", waiting until "load" +25ms
  pw:api   navigated to "https://playwright/index.html" +33ms
  pw:api   "domcontentloaded" event fired +141ms
  pw:api   "load" event fired +18ms
  pw:api   "networkidle" event fired +220ms
  pw:api   "networkidle" event fired +3ms
  pw:api   "networkidle" event fired +278ms
  pw:api   navigated to "https://public-api.wordpress.com/connect/?googleplus-sign-in=https%3A%2F%2Fe2eflowtestingnotifications.wordpress.com&color_scheme=light" +1s
  pw:api   navigated to "https://widgets.wp.com/likes/master.html?ver=20210413#ver=20210413&origin=7Cmog-p2" +105ms
  pw:api   navigated to "https://public-api.wordpress.com/wp-admin/rest-proxy/#https://widgets.wp.com" +1s
  pw:api   "domcontentloaded" event fired +74ms
  pw:api   "load" event fired +6ms
  pw:api   "domcontentloaded" event fired +3ms
  pw:api => request.response started +2ms
  pw:api   "load" event fired +4ms
  pw:api <= request.response succeeded +10ms
  pw:api <= page.waitForNavigation succeeded +1ms
  pw:api   "load" event fired +395ms
  pw:api   navigated to "https://public-api.wordpress.com/wp-admin/rest-proxy/#https://widgets.wp.com" +4ms
  pw:api   "load" event fired +43ms
  pw:api   "domcontentloaded" event fired +2ms
  pw:api   navigated to "https://widgets.wp.com/likes/index.html?ver=20210413#blog_id=112576888&post_id=64&origin=e2eflowtestingnotifications.wordpress.com&obj_id=112576888-64-610040029e0ba" +19ms
  pw:api   "domcontentloaded" event fired +8ms
  pw:api   "load" event fired +12ms
  pw:api   "networkidle" event fired +716ms
  pw:api   "load" event fired +2s
  pw:api   navigated to "https://widgets.wp.com/notifications/?v=calypso-4917-1243-gd99997ff2ev2&locale=en#p7Cmog-12-p2" +3ms
  pw:api   "domcontentloaded" event fired +159ms
  pw:api   navigated to "https://public-api.wordpress.com/wp-admin/rest-proxy/?v=2.0#https://widgets.wp.com" +59ms
  pw:api   "domcontentloaded" event fired +11ms
  pw:api   "load" event fired +1ms
  pw:api   "networkidle" event fired +970ms
```

A few things here:
- network related events fire a lot, especially for posts full of content like this one.
- a lot of navigation occur, likely related to the above fact.

According to Playwright docs for [multiple navigations](https://playwright.dev/docs/navigations/#multiple-navigations), it is good practice to explicitly wait for a specific URL in such cases.

Alternatively, the same docs state that for [complicated loading behavior](https://playwright.dev/docs/navigations/#advanced-patterns) it is recommended to use a `waitForFunction` approach.

The anticipated challenge with implementing these 'more custom' solutions is whether it will continue functioning for all situations:
- CI
- local
- staging webapp
- production webapp
- localhost webapp

Reply:
> Now, I don't know exactly why this addition resolve the issue. I suspect it might have to do with the fact that even after the page load is supposedly 'finished', the comments list continues to load new content and so the comment text area is shifted down continuously until all comments are done loading. This is just my (uneducated) hypothesis however.

Yeah, that makes sense! - while it's loading it keeps changing the element's position. I was just worried that the loading can be glitchy animation-wise which could result in premature stability assumption. Having said that, I can't think of a better way of handling that right now so let's stick with the stability check - it seems to do the job well! 🙌

Reply:
Yeah, the `stable` check isn't perfect (see my woes with Sidebar on atomic) but it seems to do well here. I'll be monitoring this test for reliability for some time.